### PR TITLE
[CentralDashboard][Licenses] Pushed license_info.csv file

### PR DESCRIPTION
**Pull request by avdaredevil**
## About
As per compliance, all open source dependencies via npm are now listed in license_info.csv file. 

## Process
First scanned with [CLI Tool](https://github.com/kubeflow/testing/tree/master/py/kubeflow/testing/node-license-tools) |
--- |
image |
Then manually added the `19` entries that failed to the end of the file |

### Meta
/area centraldashboard
/area front-end
/priority p0
/assign @avdaredevil
/cc @kwasi @zhenghuiwang

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4703)

### CODE REVIEW

File: components/centraldashboard/license_info.csv [hunk 1]
```diff
@@ -0,0 +1,1287 @@
...
+@webassemblyjs/wast-parser@1.8.5,https://github.com/xtuc/webassemblyjs/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/xtuc/webassemblyjs/master/LICENSE
+@webassemblyjs/wast-printer@1.8.5,https://github.com/xtuc/webassemblyjs/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/xtuc/webassemblyjs/master/LICENSE
+@webcomponents/shadycss@1.9.2,https://github.com/webcomponents/polyfills/tree/master/LICENSE.md,BSD-3-Clause,https://raw.githubusercontent.com/webcomponents/polyfills/master/LICENSE.md
+@webcomponents/webcomponentsjs@2.3.0,https://github.com/webcomponents/polyfills/tree/master/LICENSE.md,BSD-3-Clause,https://raw.githubusercontent.com/webcomponents/polyfills/master/LICENSE.md
+@xtuc/ieee754@1.2.0,https://github.com/feross/ieee754/tree/master/LICENSE,BSD-3-Clause,https://raw.githubusercontent.com/feross/ieee754/master/LICENSE
+@xtuc/long@4.2.2,https://github.com/dcodeIO/long.js/tree/master/LICENSE,Apache-2.0,https://raw.githubusercontent.com/dcodeIO/long.js/master/LICENSE
+JSONSelect@0.4.0,https://github.com/lloyd/JSONSelect/tree/master/LICENSE,Custom: http://jsonselect.org,https://raw.githubusercontent.com/lloyd/JSONSelect/master/LICENSE
```
Comment (zhenghuiwang) [inline components/centraldashboard/license_info.csv]: what type is `custom:` ?

File: components/centraldashboard/license_info.csv [hunk 2]
```diff
@@ -0,0 +1,1287 @@
...
+co@4.6.0,https://github.com/tj/co/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/tj/co/master/LICENSE
+code-point-at@1.1.0,https://github.com/sindresorhus/code-point-at/tree/master/license,MIT,https://raw.githubusercontent.com/sindresorhus/code-point-at/master/license
+collection-visit@1.0.0,https://github.com/jonschlinkert/collection-visit/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/jonschlinkert/collection-visit/master/LICENSE
+color-convert@0.5.3,https://github.com/harthur/color-convert/tree/master/LICENSE,MIT*,https://raw.githubusercontent.com/harthur/color-convert/master/LICENSE
+color-convert@1.9.3,https://github.com/Qix-/color-convert/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/Qix-/color-convert/master/LICENSE
+color-name@1.1.3,https://github.com/dfcreative/color-name/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/dfcreative/color-name/master/LICENSE
+colors@0.5.1,https://github.com/Marak/colors.js/tree/master/ReadMe.md,Custom: http://i.imgur.com/goJdO.png,https://raw.githubusercontent.com/Marak/colors.js/master/ReadMe.md
```
Comment (zhenghuiwang) [inline components/centraldashboard/license_info.csv]: what is `Custom: ...`?

File: components/centraldashboard/license_info.csv [hunk 3]
```diff
@@ -0,0 +1,1287 @@
...
+core-util-is@1.0.2,https://github.com/isaacs/core-util-is/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/isaacs/core-util-is/master/LICENSE
+cp-file@6.2.0,https://github.com/sindresorhus/cp-file/tree/master/license,MIT,https://raw.githubusercontent.com/sindresorhus/cp-file/master/license
+create-ecdh@4.0.3,https://github.com/crypto-browserify/createECDH/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/crypto-browserify/createECDH/master/LICENSE
+create-error-class@3.0.2,https://github.com/floatdrop/create-error-class/tree/master/license,MIT,https://raw.githubusercontent.com/floatdrop/create-error-class/master/license
+create-hash@1.2.0,https://github.com/crypto-browserify/createHash/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/crypto-browserify/createHash/master/LICENSE
+create-hmac@1.1.7,https://github.com/crypto-browserify/createHmac/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/crypto-browserify/createHmac/master/LICENSE
+cross-env@5.2.1,https://github.com/kentcdodds/cross-env/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/kentcdodds/cross-env/master/LICENS
```
Comment (zhenghuiwang) [inline components/centraldashboard/license_info.csv]: what is `Custom: ...`?

Comment (zhenghuiwang) [inline components/centraldashboard/license_info.csv]: what is `Custom: ...`?

Comment (zhenghuiwang) [inline components/centraldashboard/license_info.csv]: Unlicense? If that is the case, can we get rid of this dep?

File: components/centraldashboard/license_info.csv [hunk 4]
```diff
@@ -0,0 +1,1287 @@
...
+browserify-rsa@4.0.1,https://github.com/crypto-browserify/browserify-rsa/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/crypto-browserify/browserify-rsa/master/LICENSE
+browserify-sign@4.0.4,https://github.com/crypto-browserify/browserify-sign/tree/master/LICENSE,ISC,https://raw.githubusercontent.com/crypto-browserify/browserify-sign/master/LICENSE
+browserify-zlib@0.2.0,https://github.com/devongovett/browserify-zlib/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/devongovett/browserify-zlib/master/LICENSE
+browserslist@4.7.0,https://github.com/browserslist/browserslist/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/browserslist/browserslist/master/LICENSE
+buffer-alloc-unsafe@1.1.0,https://github.com/LinusU/buffer-alloc-unsafe/tree/master/readme.md,MIT,https://raw.githubusercontent.com/LinusU/buffer-alloc-unsafe/master/readme.md
+buffer-alloc@1.2.0,https://github.com/LinusU/buffer-alloc/tree/master/readme.md,MIT,https://raw.githubusercontent.com/LinusU/buffer-alloc/master/readme.md
```
Comment (zhenghuiwang) [inline components/centraldashboard/license_info.csv]: https://github.com/LinusU/buffer-alloc/tree/master/readme.md doesn't seem to be right. No license content found in it.

There is

https://github.com/LinusU/buffer-alloc/tree/master/LICENSE file.

File: components/centraldashboard/license_info.csv [hunk 5]
```diff
@@ -0,0 +1,1287 @@
...
+@babel/types@7.6.1,https://github.com/babel/babel/tree/master/LICENSE,MIT,https://raw.githubusercontent.com/babel/babel/master/LICENSE
+@google-cloud/monitoring@0.7.1,https://github.com/googleapis/nodejs-monitoring/tree/master/LICENSE,Apache-2.0,https://raw.githubusercontent.com/googleapis/nodejs-monitoring/master/LICENSE
+@google-cloud/promisify@0.4.0,https://github.com/googleapis/nodejs-promisify/tree/master/LICENSE,Apache-2.0,https://raw.githubusercontent.com/googleapis/nodejs-promisify/master/LICENSE
+@grpc/grpc-js@0.3.6,https://github.com/grpc/grpc-node/tree/master/LICENSE,Apache-2.0,https://raw.githubusercontent.com/grpc/grpc-node/master/LICENSE
+@grpc/proto-loader@0.4.0,https://github.com/grpc/grpc-node/tree/master/README.md,Apache-2.0,https://raw.githubusercontent.com/grpc/grpc-node/master/README.md
+@kubernetes/client-node@0.8.2,https://github.com/kubernetes-client/javascript/tree/master/LICENSE,Apache-2.0,https://raw.githubusercontent.com/kubernetes-client/javascript/master/LICENSE
+@polymer/app-layout@3.1.0,https://github.com/PolymerElements/app-layout/tree/master/README.md,BSD-3-Clause,https://raw.githubusercontent.com/PolymerElements/app-layout/master/README.md
```
Comment (zhenghuiwang) [inline components/centraldashboard/license_info.csv]: I don't see license content in the README, but
https://github.com/PolymerElements/app-layout/blob/master/package.json has the license title only.

Is it better to point to the package.json?

File: components/centraldashboard/license_info.csv (hunk 4 above)
Comment (avdaredevil) [reply components/centraldashboard/license_info.csv]: Problem is they do now, but not when the release was made (ie. the version at which this package is installed): https://github.com/LinusU/buffer-alloc/tree/a0d55382c805f3b117a055e0be203f13f2519cc5

File: components/centraldashboard/license_info.csv (hunk 3 above)
Comment (avdaredevil) [reply components/centraldashboard/license_info.csv]: It is a valid license: (similar to MIT). Let's check on it?

https://github.com/dchest/tweetnacl-js/tree/master/LICENSE

File: components/centraldashboard/license_info.csv (hunk 5 above)
Comment (avdaredevil) [reply components/centraldashboard/license_info.csv]: Fixed with global polymer license

**Comment by zhenghuiwang** [comment]
Thanks for the update:

1. can you move this file under `kubeflow/components/centraldashboard/third_party` ?

2.can you also generate the actual license.txt?  using
https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/go-license-tools/concatenate_license.py

**Pull request merged**